scheduler/scheduler.go

```diff
@@ -385,6 +406,32 @@ func (s *scheduler) startTask(id, source string) []serror.SnapError {
 		return errs
 	}
 
+	taskToPlugin := make(map[string][]core.Plugin)
+	for k := range clients {
+		cps := returnCorePlugin(clients[k])
+		taskToPlugin[k] = cps
+		mgr, err := t.RemoteManagers.Get(k)
+		// If we can't get any manager for this, no reason to continue
+		if err != nil {
+			return []serror.SnapError{serror.New(err)}
+		}
+		errs := mgr.SubscribeDeps(t.ID(), []core.Metric{}, cps)
```

Comment:
Do we need the SubscribeDeps above (line 397) or could this block satisify subscribing for local and remote deps?  If we do in fact need the above SubscribeDeps we should ensure that on a remote subscription error that the local subscriptions are also rolled back.